### PR TITLE
Formatting standards

### PR DESCRIPTION
**Pull request by MondayHopscotch**
Proposed formatting standards.

see `hxformat.json` for formatting rules

- 4 spaces instead of tabs
- curlies on same line for start

### CODE REVIEW

File: faxe/Faxe.hx [hunk 1]
```diff
@@ -6,37 +6,32 @@ package faxe;
...
 
     @:native("linc::faxe::faxe_set_debug")
-    public static function fmod_set_debug(onOff : Bool):Void;
+    public static function fmod_set_debug(onOff:Bool):Void;
 
     @:native("linc::faxe::faxe_update")
     public static function fmod_update():Void;
 
     //// Sound Banks
-
```
Comment (Tanz0rz) [inline faxe/Faxe.hx:23]: Is it possible to preserve spacing here? I am using these as section markerse.

File: faxe/FaxeSoundHelperPrivate.hx [hunk 2]
```diff
@@ -1,17 +1,17 @@
 package faxe;
 
-import faxe.FaxeEventListener.FaxeEvent;
-//TODO understand why this import looks like this
```
Comment (Tanz0rz) [inline faxe/FaxeSoundHelperPrivate.hx:4]: Is it possible to preserve import comments?

File: faxe/Faxe.hx [hunk 3]
```diff
@@ -94,86 +87,86 @@ enum abstract FmodStudioPlaybackState(Int) {
 }
 
 @:enum abstract FmodResult(Int) from Int to Int {
-    var FMOD_OK                                                 = 0;
```
Comment (Tanz0rz) [inline faxe/Faxe.hx:97]: I will just update this to the new `enum abstract` and the `Int`'s on the right will go away

File: faxe/FaxeSoundHelperPrivate.hx (hunk 2 above)
Comment (Tanz0rz) [reply faxe/FaxeSoundHelperPrivate.hx:4]: Nvm I fixed the underlying problem, so this isn't an issue

File: faxe/Faxe.hx (hunk 1 above)
Comment (MondayHopscotch) [reply faxe/Faxe.hx:23]: I haven't been able to find a setting that lets us control this kind of spacing

Comment (Tanz0rz) [reply faxe/Faxe.hx:23]: I can try some other ways to separate the sections in this file soon

File: faxe/FaxeEventListener.hx [hunk 4]
```diff
@@ -9,21 +9,21 @@ interface FaxeEventListener {
 }
 
 class FaxeCallback {
-    public static inline var CREATED:UInt                  = 1;
```
Comment (Tanz0rz) [inline faxe/FaxeEventListener.hx:12]: This part will be an issue though. I do like the preserved spacing here :(

File: README.md [hunk 5]
```diff
@@ -95,10 +95,24 @@ To play the game, run `lime test windows` in the `EZPlatformer` folder
 
 This will setup the git repo as an "installed" version of `faxe2` which can be imported by your projects the same way you import other libraries. You can see the special `dev` status when you find `faxe2` in the output of `haxelib list` and your Faxe imports in your Haxe game
 
+### Formatting
+
+This repo uses [Haxe Checkstyle](https://tanneris.me/Haxe-Checkstyle) to format all `.hx` files. The local configuration for this plugin can be found in [hxformat.json](./hxformat.json)
+
+For integrating this formatting tool into vscode, see the [Haxe Checkstyle plugin](https://tanneris.me/Haxe-Checkstyle-VSCode). Note that [formatOnSave](https://tanneris.me/Haxe-Checkstyle-Formatting) should be enabled for the best experience.
+
+## Testing This Library (currently outdated)
```
Comment (Tanz0rz) [inline README.md:104]: Remove this whole section

**Pull request merged**